### PR TITLE
Create stack.go

### PR DESCRIPTION
**Pull request by ghost**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implemented Stack in Golang using Linked List

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#399

### CODE REVIEW

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@odavhad Do the following modifications.
* for better user friendly interface, create a proper user interface which consists of taking input from the user, displaying message after each operation.
* add exit operation
* add count operation to display total number of elements present at that time.
you can take help from [this](https://github.com/TesseractCoding/NeoAlgo/pull/498/files#diff-9a5e78efddee141373b3dd717555794d)

**Comment by ghost** [comment]
@siddharth25pandey  Performed the requested modifications

- [x] for a better user-friendly interface, create a proper user interface which consists of taking input from the user, displaying message after each operation.
- [x] add exit operation
- [x] add count operation to display the total number of elements present at that time.

File: Go/README.md [hunk 1]
```diff
@@ -10,13 +10,13 @@
...
 ## Data Structures
-* [Stack](/ds/stack.go)
+* [Stack](./ds/Stack.go)
 * [Stack using Queue](./ds/stack_using_queue.go)
 * [Queue using Stack](./ds/queue_using_stack.go)
 
 ## Searching
-* [Binary Search](/search/binary_search.go)
-* [Linear Search](/search/linear_search.go)
+* [Binary Search](./search/binary_search.go)
```
Comment (siddharth25pandey) [inline Go/README.md]: remove this line

File: Go/README.md [hunk 2]
```diff
@@ -10,13 +10,13 @@
...
+* [Linear Search](./search/linear_search.go)
```
Comment (siddharth25pandey) [inline Go/README.md]: Remove this line also.
As both lines show different issues details

**Review by siddharth25pandey — APPROVED** [review]
@odavhad Looks fine to me. Just do the changes as suggested
Able to merge.
Thanks

**Review by iamrajiv — APPROVED** [review]
@odavhad LGTM! hope the code is formatted with gofmt.

**Comment by ghost** [comment]
> @odavhad LGTM! hope the code is formatted with gofmt.

yes, it is.

**Review by atarax665 — APPROVED** [review]
LGTM :)

**Pull request merged**